universe/supplycommit/transitions.go

```diff
@@ -984,6 +984,11 @@ func (c *CommitFinalizeState) ProcessEvent(event Event,
 				"dangling updates: %w", err)
 		}
 
+		// We know our tree has been updated, so we need to make sure
+		// the negative lookup cache of the ignore checker is flushed
+		// and the new ignore leaves are loaded from disk.
+		env.IgnoreCheckerCache.InvalidateCache()
```

Comment:
Couldn't we just _selectively_ invalidate the cache for this particular group key? 

Even before that, we can check to see if this batch adds any new ignored assets, only invalidating if so.